lib/utils/extensions/payment_extensions.dart

```diff
@@ -1,34 +1,63 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
 import 'package:breez_sdk/bridge_generated.dart';
 import 'package:breez_translations/generated/breez_translations.dart';
 import 'package:c_breez/utils/extensions/breez_pos_message_extractor.dart';
+import 'package:fimber/fimber.dart';
+import 'package:flutter/material.dart';
+
+final _log = FimberLog("PaymentExtensions");
+
+class PaymentExtensions {
+  final Payment _payment;
+  final BreezTranslations _texts;
+  final Map<String, dynamic> _metadataMap = {};
 
-extension PaymentExtensions on Payment {
```

Comment:
@ademar111190  I certainly agree we need to handle all these details out of the widget and thanks for initiating this! 
This complexity makes me think we need another layer for `PaymentItem` and this should be on the bloc level rather than on the UI layer.
I would go here for a slightly bigger change as I think it worth it:
At the bloc level create a `PaymentItem` model for the UI (similar to this PaymentExtensions) and emit it instead of the sdk.Payment: https://github.com/breez/c-breez/blob/main/lib/bloc/account/account_state_assembler.dart#L32

I think also all the parsing should be at the constructor initializing fields or properties and then in the UI just render the values right away.

Reply:
@roeierez I just updated the PR with the following strategy; let me know what you think:

- The UI will use a model that I named PaymentMinutiae instead of the SDK payment; this model holds the data already formatted and ready for the UI
- The account state assembler is making the conversion, i.e., It is being made by the bloc. Still, the details of the conversion I'm keeping internally on `_PaymentMinutiaeFactory` accessed through factory `PaymentMinutiae.fromPayment` to make tests easier and avoid congestion on the account state assembler file as it will grow in the future.
- I kept the payment id on the payment minutiae in case of need to access the payment model back at some point in the flow.